code/modules/mob/living/living_defense.dm

```diff
@@ -91,43 +91,77 @@
 /mob/living/proc/is_ears_covered()
 	return null
 
-/mob/living/bullet_act(obj/projectile/hitting_projectile, def_zone, piercing_hit = FALSE)
+/mob/living/bullet_act(obj/projectile/proj, def_zone, piercing_hit = FALSE)
 	. = ..()
-	if(. != BULLET_ACT_HIT)
-		return .
-	if(!hitting_projectile.is_hostile_projectile())
-		return BULLET_ACT_HIT
+	var/blocked = check_projectile_armor(def_zone, proj)
 
-	// we need a second, silent armor check to actually know how much to reduce damage taken, as opposed to
-	// on [/atom/proc/bullet_act] where it's just to pass it to the projectile's on_hit().
-	var/armor_check = check_projectile_armor(def_zone, hitting_projectile, is_silent = TRUE)
+	if(blocked >= 100)
+		if(. == BULLET_ACT_HIT && proj.is_hostile_projectile())
+			apply_projectile_effects(proj, def_zone, blocked)
+		return .
 
+	var/turf/our_turf = get_turf(src)
+	var/hit_limb_zone = check_hit_limb_zone_name(def_zone)
+	var/obj/item/bodypart/hit_bodypart = get_bodypart(hit_limb_zone)
+	if (proj.damage && proj.damage_type == BRUTE)
```

Comment:
I think all this blood splatter stuff needs to check for ` . == BULLET_ACT_HIT`


In parent all bullet act, `COMSIG_ATOM_PRE_BULLET_ACT` can return `BULLET_ACT_FORCE_PIERCE` or `BULLET_ACT_BLOCK`. 

In current code, this means no blood splatter is created (as `on_hit` never runs), and no damage is dealt (as `. != BULLET_ACT_HIT`). 

But in this implementation, it will spawn a blood splatter (as it doesn't care about parent ret val), and then... deal no damage.

Reply:
Also, it would be nice if this was put in `apply_projectile_effects` in a way, to match convention with the attack chain: 

`item/attack` -> `living/attacked_by` (damage dealt here) -> `living/attack_effects` (blood spawned here)